MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,279 @@
+            #. x (any): The returned result(s) of run_step
+        """
+        pass
+
+    def finalize(self):
+        """
+        Finalizes the calculations (e.g. averaging the total term, output files creations ...).
+        """
+
+        # Get filter class and instantiate filter object
+        filter_config = json.loads(self.configuration["trajectory_filter"]["value"])
+
+        filter_class, filter_attributes = (
+            FILTER_MAP[filter_config["filter"]],
+            filter_config["attributes"],
+        )
+
+        if {"n_steps", "time_step_ps"} not in set(filter_attributes.keys()):
+            filter_attributes.update(
+                {
+                    "n_steps": self.configuration["trajectory"]["length"],
+                    "time_step_ps": self.configuration["trajectory"]["md_time_step"],
+                }
+            )
+
+        filter = filter_class(**filter_attributes)
+
+        trajectories = copy.deepcopy(self.atomic_trajectory_array)
+
+        # Magnitude of zero frequency in filter response (equivalent to the average atomic positions)
+        zero_magnitude = np.abs(filter.freq_response.magnitudes[0])
+
+        # Apply filter (only apply initial position offset to atoms if filter response f(0) != 1)
+        filtered_coords = apply(
+            filter,
+            trajectories,
+            apply_offsets=not np.isclose(zero_magnitude, 1),
+        )
+
+        # Create new chemical system for output trajectory
+        name = self.configuration["output_files"]["file"].name.split(".").pop(0)
+        if not isinstance(name, str):
+            name = "filtered_traj_chemical_system"
+        output_chemical_system = ChemicalSystem(name)
+        output_chemical_system.initialise_atoms(self._selected_atoms)
+
+        # Create trajectory writer object
+        self._output_trajectory = TrajectoryWriter(
+            self.configuration["output_files"]["file"],
+            output_chemical_system,
+            filter_attributes["n_steps"],
+            None,
+            positions_dtype=self.configuration["output_files"]["dtype"],
+            compression=self.configuration["output_files"]["compression"],
+        )
+
+        # Write trajectory
+        write_filtered_trajectory(
+            parent_configuration=self.configuration,
+            nsteps=filter_attributes["n_steps"],
+            filtered_coordinates=filtered_coords,
+            output_trajectory=self._output_trajectory,
+        )
+
+        # The input trajectory is closed.
+        self.configuration["trajectory"]["instance"].close()
+
+        # The output trajectory is closed.
+        self._output_trajectory.close()
+
+        # Write the filter metadata to output
+        outputFile = h5py.File(self.configuration["output_files"]["file"], "r+")
+        outputFile.create_group("metadata").create_dataset(
+            "trajectory_filter",
+            (1,),
+            data=filter.__str__(),
```

Comment:
```suggestion
            data=str(filter),
```
Shouldn't need to call dunders